mysql/tls-reload/main.go

```diff
@@ -0,0 +1,190 @@
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Wait for an os.Interrupt signal
+	go waitForCtrlC(cancel)
+
+	// Start X.509 watcher
+	startWatcher(ctx)
+}
+
+func startWatcher(ctx context.Context) {
+	// Creates a new Workload API client, connecting to provided socket path
+	// Environment variable `SPIFFE_ENDPOINT_SOCKET` is used as default
+	client, err := workloadapi.New(ctx, workloadapi.WithAddr(socketPath))
+	if err != nil {
+		log.Fatalf("Unable to create workload API client: %v", err)
+	}
+	defer client.Close()
+
+	// Start a watcher for X.509 SVID updates
+	doneCh := make(chan struct{}, 1)
+	go func() {
+		err := client.WatchX509Context(ctx, &x509Watcher{})
+		if err != nil && status.Code(err) != codes.Canceled {
+			log.Fatalf("Error watching X.509 context: %v", err)
+		}
+		doneCh <- struct{}{}
+	}()
+
+	<-doneCh
+}
+
+// x509Watcher is a sample implementation of the workloadapi.X509ContextWatcher interface
+type x509Watcher struct{}
+
+// OnX509ContextUpdate is run every time an SVID is updated
+func (w *x509Watcher) OnX509ContextUpdate(c *workloadapi.X509Context) {
+	// write SVID to disk
+	err := writeX509Context(c)
+	if err != nil {
+		log.Printf("Failed to write SVID/Bundle to disk: %v", err)
+		return
+	}
+
+	log.Printf("Successfully written SVID/Bundle to disk")
+
+	// Create TLS config with client certificates
+	tlsConf, err := createTLSConf(c)
+	if err != nil {
+		log.Printf("Failed to create MySQL TLS config: %v", err)
+		return
+	}
+
+	err = mysql.RegisterTLSConfig("spire-ssl", tlsConf)
+	if err != nil {
+		log.Printf("Failed to register MySQL TLS config: %v", err)
+		return
+	}
+
+	db, err := sql.Open("mysql", dbConnectionString)
+	if err != nil {
+		log.Printf("Failed to open MySQL database: %v", err)
+		return
+	}
+	defer db.Close()
+
+	_, err = db.ExecContext(context.Background(), reloadTLSQuery)
+	if err != nil {
+		log.Printf("Failed to run reload TLS query: %v", err)
+		return
+	}
+
+	log.Printf("Successfully reloaded MySQL TLS config")
+}
+
+func writeX509Context(c *workloadapi.X509Context) error {
+	svid, err := getSVIDByHint(c, mysqlServerSVIDHint)
+	if err != nil {
+		return err
+	}
+
+	certBytes, keyBytes, err := svid.Marshal()
+	if err != nil {
+		return err
+	}
+
+	bundleBytes, err := c.Bundles.Bundles()[0].Marshal()
+	if err != nil {
+		return err
+	}
+
+	if err = os.WriteFile(certFilePath, certBytes, 0o644); err != nil {
+		return err
+	}
+
+	if err = os.WriteFile(keyFilePath, keyBytes, 0o644); err != nil {
+		return err
+	}
+
+	if err = os.WriteFile(bundleFilePath, bundleBytes, 0o644); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createTLSConf(c *workloadapi.X509Context) (*tls.Config, error) {
+	svid, err := getSVIDByHint(c, mysqlClientSVIDHint)
+	if err != nil {
+		return nil, err
+	}
+
+	return tlsconfig.MTLSClientConfig(svid, c.Bundles.Bundles()[0], tlsconfig.AuthorizeID(mysqlServerSPIFFEID)), nil
```

Comment:
```suggestion
	return tlsconfig.MTLSClientConfig(svid, c.Bundles, tlsconfig.AuthorizeID(mysqlServerSPIFFEID)), nil
```
I think this should also work, a little simpler